src/impression.js

```diff
@@ -235,3 +241,65 @@ function applyResponse(win, response) {
     win.history.replaceState(null, '', newHref);
   }
 }
+
+/**
+ * Return a promise that whether appending extra url params to outgoing link is required.
+ * @param {!Window} win
+ * @return {!Promise<boolean>}
+ */
+export function shouldAppendExtraParams(win) {
```

Comment:
Why does this go over window and not ampdoc? Last time we had issues with this for PWA.

Reply:
Hmm i thought it's basically doing `ampdoc.win` in this function or in caller function. I didn't realize that there're differences here. I changed it to ampdoc